src/modules/flow/iio/nodes.c

```diff
@@ -25,34 +25,42 @@
 
 #include <sol-iio.h>
 
-struct iio_direction_vector_data {
+struct iio_device_config {
     struct sol_iio_config config;
-    struct sol_direction_vector scale;
-    struct sol_direction_vector offset;
     struct sol_drange_spec out_range;
     struct sol_iio_device *device;
-    struct sol_iio_channel *channel_x;
-    struct sol_iio_channel *channel_y;
-    struct sol_iio_channel *channel_z;
     bool buffer_enabled : 1;
     bool use_device_default_scale : 1;
     bool use_device_default_offset : 1;
 };
 
-struct color_data {
-    struct sol_iio_config config;
+struct iio_channel_config {
     double scale;
     double offset;
+    struct sol_iio_channel *iio_channel;
+};
+
+struct iio_direction_vector_data {
+    struct sol_iio_config config;
+    struct sol_direction_vector scale;
+    struct sol_direction_vector offset;
     struct sol_drange_spec out_range;
     struct sol_iio_device *device;
-    struct sol_iio_channel *channel_red;
-    struct sol_iio_channel *channel_green;
-    struct sol_iio_channel *channel_blue;
+    struct sol_iio_channel *channel_x;
+    struct sol_iio_channel *channel_y;
+    struct sol_iio_channel *channel_z;
     bool buffer_enabled : 1;
     bool use_device_default_scale : 1;
     bool use_device_default_offset : 1;
 };
 
+struct color_data {
+    struct iio_channel_config channel_red;
+    struct iio_channel_config channel_green;
+    struct iio_channel_config channel_blue;
+    struct iio_device_config iio_config;
+};
```

Comment:
Guess I've got it: because there's no 'RGB' packet to set scale and offset. But then`iio_channel_config` is a misleading name, maybe `color_channel_config`?